### PR TITLE
feat: Support for k8s 1.23 and statuscake's updated api

### PR DESCRIPTION
**Pull request by MuneebAijaz**
This PR takes care of the following issues/features:

1. IMC was degraded on new versions of k8s due to deprecated ingress v1beta1 version.
2. Statuscake migrated their api, in this PR, implementation of new api has been taken care of.

### CODE REVIEW

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/cc72ddaad000a48e6c7bed3246558ab11b39a6cd/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/a114d334c4d26e8d8a33d19062ecb07065a3ce62/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/911340156ab53ff566a57efe19b3026fbd7dec2f/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/e92058c1b8c57a2391d806192184416a4faee5e1/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/35cbf2dc17553c5f5f233cba27f99566aace281e/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/edf639892c5f216306ba76fdcc33c42be054509b/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/e05897d52238fc0b9dfc358e43481e3db805a39e/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/27ebcdcc2d5560c4912c64c2051d63e2d24d8e66/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/21f36c337eebc999daa40c8e38b35c40e4285caf/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/b18a87802830d43f12209160ff4ad49cd1a3d2fc/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Image is available for testing. `docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-398-b18a8780`

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/576cf400a203c333e6545229547131e8658dd7ee/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Image is available for testing. `docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-398-576cf400`

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/ac17a5256d580028849507df836f11323d8eabe7/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/ed87c8c5380ad6c84bc365b40ed9a888869a380b/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/846322b661ec9efd60efcbaadf88298ae7e1b63c/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Image is available for testing. `docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-398-846322b6`

File: examples/configs/test-config-statuscake.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 providers:
   - name: StatusCake
     apiKey: API_KEY
-    apiURL: https://app.statuscake.com/API/
+    apiURL: https://api.statuscake.com/v1/uptime
     username: USERNAME
 enableMonitorDeletion: true
```
Comment (faizanahmad055) [inline examples/configs/test-config-statuscake.yaml:6]: Please add endline

Comment (faizanahmad055) [inline examples/configs/test-config-statuscake.yaml:6]: Add endline

**Comment by stakater-user** [comment]
@MuneebAijaz Image is available for testing. `docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-398-1d962658`

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/733ba1c6a76dda5b11688fb6660151b480f05df4/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/1500da01ec33c97f85f7a974d4fedabca839c4db/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/4a18d2fcf018e6d82bedfb09700279e754795194/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/27212452c5ed5161d281dbb3e894c9c8ec0274f0/checks) has Failed!

**Comment by stakater-user** [comment]
@MuneebAijaz Image is available for testing. `docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-398-508c1a57`

File: pkg/monitors/statuscake/statuscake-monitor.go [hunk 2]
```diff
@@ -210,37 +234,90 @@ func (service *StatusCakeMonitorService) GetByName(name string) (*models.Monitor
...
+
 	if resp.StatusCode == http.StatusOK {
-		f := make([]StatusCakeMonitorMonitor, 0)
-		err := json.NewDecoder(resp.Body).Decode(&f)
+		var StatusCakeMonitor StatusCakeMonitor
+		var StatusCakeMonitorData []StatusCakeMonitorData
+		err = json.Unmarshal(bodyBytes, &StatusCakeMonitor)
 		if err != nil {
-			log.Error(err, "Unable to retrieve monitor")
+			log.Error(err, "Unable to unmarshal response")
```
[the remark this answers is not in the document]
Comment (MuneebAijaz) [reply pkg/monitors/statuscake/statuscake-monitor.go]: updated

File: examples/configs/test-config-statuscake.yaml (hunk 1 above)
Comment (MuneebAijaz) [reply examples/configs/test-config-statuscake.yaml]: done

**Review by tanalam2411 — APPROVED** [review]

**Pull request merged**